Выполнено задание ФЗ 6(Надеюсь правильно)

LUKIDMI opened the pull request:
В InMemoryHistoryManager добавлены:
Реализация метода void remove(int id);
Метод void linkLast(Task task);
Метод void removeNode(Node node);
Добавлен класс Node
Написаны тесты

LUKIDMI committed:
Выполнено задание ФЗ 6(Надеюсь правильно)
В InMemoryHistoryManager добавлены:
Реализация метода void remove(int id);
Метод void linkLast(Task task);
Метод void removeNode(Node node);
Добавлен класс Node
Написаны тесты

LUKIDMI committed:
Выполнены требования по кодстайлу из тестов гитхаба

LUKIDMI committed:
Выполнены требования по кодстайлу из тестов гитхаба, finally убран пробел

LUKIDMI committed:
Убрал ненужный импорт в SubTaskTest

LUKIDMI committed:
Добавлен пробел перед фигурной скобкой в TaskTest

LUKIDMI committed:
Поправил отступы в InMemoryHistoryManager и InMemoryTaskManager

LUKIDMI committed:
Добавлены отступы в интерфейс HistoryManager

LUKIDMI committed:
Переместил папку test в корень проекта

LUKIDMI committed:
Разбираюсь с ошибками в тестах

LUKIDMI committed:
Исправлены тесты

LUKIDMI committed:
Добавлены отступы в InMemoryHistoryManager

irek-valiev commented on src/pavel/kalinkin/project/manager/InMemoryHistoryManager.java:
```diff
@@ -2,25 +2,98 @@
+        }
+        linkLast(task);
+    }
+
```
добавь явно аннотацию @Override

LUKIDMI replied:
Добавил

irek-valiev commented on src/pavel/kalinkin/project/manager/InMemoryHistoryManager.java:
```diff
@@ -2,25 +2,98 @@
 
+    private void linkLast(Task task) {
+        int taskId = task.getId();
+        Node oldLast = lastNode;
```
добавь переменной oldLast модификатор final, чтобы дальнейшие изменения в не пробрасывались

LUKIDMI replied:
Добавил

irek-valiev commented on src/pavel/kalinkin/project/manager/InMemoryTaskManager.java:
```diff
@@ -206,6 +215,10 @@ public List<SubTask> getAllSubTasks() {
+            for (Integer subTaskId : epic.getEpicSubTasks()) {
+                historyManager.remove(subTaskId);
+            }
+            historyManager.remove(epic.getId());
```
эпики из истории удалять не нужно

LUKIDMI replied:
Исправил

LUKIDMI committed:
Исправлены ошибки после ревью
Исправлен метод deleteAllSubtasks в InMemoryTaskManager
Добавлена аннотация к реализации метода remove
Добавлен модификатор final к oldLast в методе linkLast
Спасибо большое, Ирек!
